sycl/include/sycl/group.hpp

```diff
@@ -353,12 +354,13 @@ template <int Dimensions = 1> class __SYCL_TYPE(group) group {
   /// Permitted types for DestDataT are all scalar and vector types, except
   /// boolean. SrcDataT must be either the same as DestDataT or const DestDataT.
   template <typename DestDataT, typename SrcDataT>
-  std::enable_if_t<!detail::is_bool<DestDataT>::value &&
-                       std::is_same_v<std::remove_const_t<SrcDataT>, DestDataT>,
-                   device_event>
+  static std::enable_if_t<
```

Comment:
This is an ABI change. Which makes it not spec compliant. Can we make group stateless too so that this isn't needed?

Reply:
> Can we make group stateless too so that this isn't needed?

I'm not sure about that, but I'll revert group class changes.

> This is an ABI change.

Are these methods part of the SYCL runtime ABI?

Reply:
> Are these methods part of the SYCL runtime ABI?

Sorry. I meant to say API. The spec doesn't define an ABI.
To be specific: The SYCL spec says that the methods aren't static. And it is an user-observable difference whether they are or aren't static.

Reply:
Reverted.